coloredlettrine.dtx

```diff
@@ -192,9 +209,45 @@
 % \end{macro}
 %
 %    \begin{macrocode}
-\lettrine[#1]{%
-\rlap{\color{\EBLettrineBackColor} #2}%
-     {\EBLettrineFrontFont\color{\EBLettrineFrontColor} #2}}{#3}%
+\iffirstLineFilled
+  \renewcommand{\afterInitial}{\makebox[\firstLineWidth][s]{#3}}
+\else
+  \renewcommand{\afterInitial}{#3}
+\fi
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+\def\givenString{#2}
+\StrLen{\givenString}[\numOfChars]
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+\ifnum\numOfChars>1%
+    \coloredlettrinewithquote{#1}{#2}{\afterInitial}%
+\else%
+    \lettrine[#1]{%
+        \rlap{\color{\EBLettrineBackColor} #2}%
+        {\EBLettrineFrontFont\color{\EBLettrineFrontColor} #2}%
+    }{\afterInitial}%
+\fi
+}%
+%    \end{macrocode}
+%    \begin{macrocode}
+\newcommand{\clrlet}{\coloredlettrine}
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+\newcommand{\coloredlettrinewithquote}[3]{%
```

Comment:
This looks like it is intended as a public macro.

Could you maybe add a description for it in the `Usage` section (after line 106)?

Reply:
`coloredlettrinewithquote` is not (if you mean by that, that the user calls it), but others are. I will add descriptions for those. Also, the name _...withquote_ is not really correct any more because there are other hanging punctuations now, so I will rename it.

Reply:
Great 👍

Reply:
I couldn't generate the documentation. I changed `\CheckSum{30}` in line 49 to `\CheckSum{0} \MaybeStop` and it works now. Is that okay and should I commit that?

Reply:
Yes, afaik this is the way to do it.